test/test_binary_ufuncs.py

```diff
@@ -2011,6 +2011,58 @@ def test_bitwise_xor(self, device):
                          torch.bitwise_xor(torch.tensor([True, True, False], device=device),
                                            torch.tensor([False, True, False], device=device)))
 
+    def test_bitwise_left_shift(self, device):
```

Comment:
Let's create "test_shift" and do the following:

- use the @dtypes decorator and select dtypes torch.uint8, torch.int64
- enumerate pairs (torch_op, numpy_op) that define the functions to test
- the pairs should be (torch.bitwise_left_shift, np.left_shift), (operator.left_shift, operator.left_shift), (torch.bitwise_right_shift, np.bitwise_right_shift), (operator.right_shift, operator.right_shift)
- compare the results on a variety of test cases that use the test's device and dtype
- don't bother testing the out or inplace variants, those are tested automatically when creating an opinfo for torch.bitwise_left_shift and torch.bitwise_right shift

See https://docs.python.org/3/library/operator.html for more on using the operator module.

fyi @pmeier, this highlights the importance of automated binary elementwise testing, and @kshitij12345, I expect we'll increasingly be interested in supporting operators using OpInfos (operators are like aliases in that operator.left_shift is an alias for torch.bitwise_left_shift, but there are no method variants for operators and the inplace operator variants are different, because the inplace version of the operator is operator.ilshift, so we'll need to think of the appropriate way to model them -- maybe as totally different ops with their own OpInfos?)

FYIs aside, we're not done with testing yet, @asi1024! We also need to add a test that validates PyTorch's type promotion behavior and support for floating point inputs. For this test we can just "spot check" one input for each case we want to validate. In particular, we should check:

- int tensor x float scalar == int tensor x trunc(float scalar) type promotion
- float tensor x int scalar works as expected (it's OK to use a 'golden value' here)
- float tensor x float scalar works as expected (it's OK to use a 'golden value' here)

Does that make sense, @asi1024? I know that's a lot, but we should be very careful validating the behavior here.

Reply:
I updated the tests in c35027316aea7fa1d04d53516f64dfb3c99147d6!